service/controller/rolebindingtemplate/resource/rolebinding/create.go

```diff
@@ -34,11 +35,14 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 			return microerror.Mask(err)
 		}
 
-		if err = rbac.CreateOrUpdateRoleBinding(r, ctx, ns, roleBinding); err != nil {
-			r.logger.Debugf(ctx, "Could not apply roleBinding %s to namespace %s due to error %v", roleBinding.Name, ns, err)
-			continue
+		roleBinding = cleanSubjects(roleBinding, ns)
+		if len(roleBinding.Subjects) > 0 {
```

Comment:
we go through the subjects and remove the ones not fitting the restrictions if it's a protected namespace. We still create it in case subjects are left